cmd/flux/test.go

```diff
@@ -52,26 +52,37 @@ func (testExecutor) Run(pkg *ast.Package) error {
 
 	var output strings.Builder
 	results := flux.NewResultIteratorFromQuery(query)
+
+	foundErrorResult := false
 	for results.More() {
 		result := results.Next()
-		err := result.Tables().Do(func(tbl flux.Table) error {
-			// The data returned here is the result of `testing.diff`, so any result means that
-			// a comparison of two tables showed inequality. Capture that inequality as part of the error.
-			// XXX: rockstar (08 Dec 2020) - This could use some ergonomic work, as the diff output
-			// is not exactly "human readable."
-			_, _ = fmt.Fprint(&output, table.Stringify(tbl))
-			return nil
-		})
-		if err != nil {
-			return err
+		if result.Name() == "error" {
+			foundErrorResult = true
+
+			err := result.Tables().Do(func(tbl flux.Table) error {
+				// The data returned here is the result of `testing.diff`, so any result means that
+				// a comparison of two tables showed inequality. Capture that inequality as part of the error.
+				// XXX: rockstar (08 Dec 2020) - This could use some ergonomic work, as the diff output
+				// is not exactly "human readable."
+				_, _ = fmt.Fprint(&output, table.Stringify(tbl))
+				return nil
+			})
```

Comment:
It would be good to run `Tables.Do()` and then `tbl.Do()` on the tables in the other results, I can't think of a scenario where it would cause a deadlock not to, but it is good practice.